pilot/pkg/features/experimental.go

```diff
@@ -197,4 +197,8 @@ var (
 	EnableLocalityWeightedLbConfig = env.Register("ENABLE_LOCALITY_WEIGHTED_LB_CONFIG", false,
 		"If enabled, always set LocalityWeightedLbConfig for a cluster, "+
 			" otherwise only apply it when locality lb is specified by DestinationRule for a service").Get()
+
+	EnableEnhancedDestinationRuleMerge = env.Register("ENABLE_ENHANCED_DESTINATIONRULE_MERGE", true,
```

Comment:
This doesn't seem to fully apply, it looks like we only check at some points. I see the same behavior with and without the flag following the reproducer case in https://github.com/istio/istio/issues/52519

Reply:
Weired, let me have a check with that, maybe later